### ISSUE TITLE
Is there any documentation on sqlalchemy.testing?

### ISSUE THREAD

**Issue by darthmall** [sqlalchemy/sqlalchemy#4998]
It looks like the `sqlalchemy.testing` package contains utilities for writing tests, but I'm not able to find any documentation on how to use it. I've read through some of the code in the module as well as looked at a few SQLAlchemy's own tests, but there's a lot in there and I'm not sure where the best place to start might be.

Is there a guide on testing apps that use SQLAlchemy or reference documentation that I'm just not finding? (Bonus points if it also uses Alembic to run migrations for standing up the schema in the database.)

**Comment by gordthompson** [comment]
Have you seen this?

https://github.com/sqlalchemy/sqlalchemy/blob/master/README.unittests.rst

**Comment by zzzeek** [comment]
oh, and you wouldn't use sqlalchemy.testing for that.    I think that's what's confusing here, how did you come across sqlalchemy.testing ?   it's used only for third party dialects and the README linked above explains its purpose.

**Comment by darthmall** [comment]
> oh, and you wouldn't use sqlalchemy.testing for that. I think that's what's confusing here, how did you come across sqlalchemy.testing ? it's used only for third party dialects and the README linked above explains its purpose.

I had a suspicion that might be the case. Thanks. 🙏 

I had been searching the web for guides on testing SQLAlchemy projects and hadn't turned up anything that was useful, so I started reading through SQLAlchemy's tests to see if there were any hints in there as to how I should set up my testing infrastructure; that's where I came across `sqlalchemy.testing`.
